packages/AdaptiveCards/Actions/SendActionExecuteResponse.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using AdaptiveCards.Templating;
+using AdaptiveExpressions.Properties;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Builder.Dialogs;
+using Microsoft.Bot.Schema;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Bot.Components.AdaptiveCards
+{
+    public class SendActionExecuteResponse : Dialog
+    {
+        [JsonConstructor]
+        public SendActionExecuteResponse([CallerFilePath] string sourceFilePath = "", [CallerLineNumber] int sourceLineNumber = 0)
+            : base()
+        {
+            // enable instances of this command as debug break point
+            this.RegisterSourceLocation(sourceFilePath, sourceLineNumber);
+        }
+
+        [JsonProperty("$kind")]
+        public const string Kind = "Microsoft.Bot.Components.SendActionExecuteResponse";
+
+        [JsonProperty("disabled")]
+        public BoolExpression Disabled { get; set; }
+
+        [JsonProperty("statusCode")]
+        public IntExpression StatusCode { get; set; } = new IntExpression(200);
+
+       [JsonConverter(typeof(StringEnumConverter), /*camelCase*/ true)]
+        public enum ValueType
```

Comment:
Since the enum is public, this should go into a separate file.

Reply:
This is how the SDK defines enums for actions. I think they're required to be subtypes for serialization purposes. Take it up with Tom & Carlos.

---
In reply to: [610217865](https://github.com/microsoft/botframework-components/pull/862#discussion_r610217865) [](ancestors = 610217865)